src/pages/Projects/components/RoleModal/RoleModal.tsx

```diff
@@ -174,6 +177,47 @@ export default function RoleModal({
 
   const roleAssignments = watch("assignments");
 
+  const selectedSkill = watch("skillId");
+
+  useEffect(() => {
+    if (selectedSkill) {
+      filterEmployees(selectedSkill);
+    }
+  }, [selectedSkill]);
+
+  useEffect(() => {
+    // When the list of employees changes, we need to delete any added assignment
+    // that has an inadequate employee
+    // and we only do it if we're in Add role mode
+    if (roleAssignments && !roleToEdit) {
+      resetAssignments(filteredEmployees, roleAssignments);
+    }
+  }, [filteredEmployees]);
+
+  const filterEmployees = (skillId: string) => {
+    setFilteredEmployees(
+      employees.filter((employee) => employee.skills[0].id == skillId),
```

Comment:
shouldn't this be looping over all `employee.skills`?

Reply:
the last I heard we only check the first skill because we only assign 1 skill per person for now, but that may have changed

Reply:
Oh, I dunno any history. All I know is that you _can_ assign multiple skills to an employee currently:
<img width="1142" alt="image" src="https://user-images.githubusercontent.com/92131113/179827390-f6689ea0-d077-45ec-9646-0db6e63dd8fb.png">

Hoping someone answers your question in slack :)